src/Microsoft.Health.Fhir.Core/Features/Search/Converters/MoneyToQuantitySearchValueConverter.cs

```diff
@@ -27,7 +27,7 @@ protected override IEnumerable<ISearchValue> Convert(ITypedElement value)
         {
             var decimalValue = (decimal?)value.Scalar("value");
 
-            if (decimalValue == null)
+            if (!decimalValue.HasValue)
```

Comment:
Will this work if it is null? This looks like it would cause a null pointer exception.

Reply:
It's a decimal? -> Nullable<decimal> which we assign line above.
It can hold null, but it's struct, not a class, so their can't be null by definition.

---
In reply to: [617664792](https://github.com/microsoft/fhir-server/pull/1858#discussion_r617664792) [](ancestors = 617664792)